zamba/object_detection/yolox/megadetector_lite_yolox.py

```diff
@@ -116,19 +122,63 @@ def _preprocess(self, frame: np.ndarray) -> np.ndarray:
         """Process an image for the model, including scaling/padding the image, transposing from
         (height, width, channel) to (channel, height, width) and casting to float.
         """
-        return np.ascontiguousarray(
-            self.scale_and_pad_array(
-                frame, self.config.image_width, self.config.image_height
-            ).transpose(2, 0, 1),
+        arr = np.ascontiguousarray(
+            self.scale_and_pad_array(frame, self.config.image_width, self.config.image_height),
             dtype=np.float32,
         )
+        return np.moveaxis(arr, 2, 0)
+
+    def _preprocess_video(self, video: np.ndarray) -> np.ndarray:
+        """Process a video for the model, including resizing the frames in the video, transposing
+        from (batch, height, width, channel) to (batch, channel, height, width) and casting to float.
+        """
+        resized_frames = []
+        for frame_idx in range(video.shape[0]):
+            frame = video[frame_idx]
+            resized_frames.append(self._preprocess(frame))
+
+        return np.array(resized_frames)
+
+    def detect_video(self, video_arr: np.ndarray, pbar: bool = False):
+        """Runs object detection on an video.
+
+        Args:
+            video_arr (np.ndarray): An video array with dimensions (frames, height, width, channels).
+
+        Returns:
+            list: A list containing detections and score for each frame. Each tuple contains two arrays:
+                the first is an array of bounding box detections with dimensions (object, 4) where
+                object is the number of objects detected and the other 4 dimension are
+                (x1, y1, x2, y1). The second is an array of object detection confidence scores of
+                length (object) where object is the number of objects detected.
+        """
 
-    def detect_video(self, frames: np.ndarray, pbar: bool = False):
         pbar = tqdm if pbar else lambda x: x
 
+        # iterate over batches of 24
+        batch_size = 24
+
+        video_outputs = []
+        with torch.no_grad():
+            for i in range(0, len(video_arr), batch_size):
+                a = video_arr[i : i + batch_size]
```

Comment:
I assume that we tested and this works fine if the len of the array is not divisible by the batch size?

Reply:
yep! for example, array with len 25 and batch size of 24 succeeds